### ISSUE TITLE
Do not allow to subclass builtins

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#600]
# Rule request

## Thesis

We should disallow things like:

```python
class MyDict(dict): ...

class SuperInt(int): ...
```

## Reasoning

It is better to write a wrapper:

```python
class MyDictWrapper(object):
     def __init__(self):
          self._inner_dict = {}
```

Or to implement several methods to make your API native: `Number(1) + Number(2) == Number(3)`

Related: https://github.com/troessner/reek/blob/master/docs/Subclassed-From-Core-Class.md
Original: https://github.com/wemake-services/wemake-python-styleguide/issues/422

**Comment by sobolevn** [comment]
* Except `object`, it can and should be subclassed.

**Comment by sobolevn** [comment]
And `type` is fine.
